### PR TITLE
Add open modal and source http/https errors

### LINKED ISSUE #374: Source without https doesn’t get warning
As one of the survey respondents pointed out, “source without https should get warning”.

As @orangemug pointed out, we can probably use https://www.mapbox.com/mapbox-gl-js/api/#map.event:error here and print any errors that occur when loading resources.

### PR DESCRIPTION
**Pull request by orangemug**
If the editor is loaded from `https://` (like on `https://maputnik.github.io/editor/`) show an error for url fields where the user is trying to use `http://` urls 

Demo: <https://1815-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html>

Screen Shot 2019-10-27 at 17 47 54
Screen Shot 2019-10-27 at 17 48 12

Fixes #374

### CODE REVIEW

**Comment by pathmapper** [comment]
Very nice @orangemug, should this work also for 
- Sprite URL (_Style Settings_)
- Glyphs URL (_Style Settings_)
- GeoJSON (URL) (_Data Sources_) ?

**Comment by orangemug** [comment]
Yeah @pathmapper you're right, added that in, demo: https://1854-84182601-gh.circle-artifacts.com/0/artifacts/build/index.html#0.35/0/0

File: src/components/inputs/UrlInput.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  };
+  const protocol = getProtocol(url);
+  if (
+    protocol &&
+    protocol === "http:" &&
+    window.location.protocol === "https:"
+  ) {
+    error = (
+      <SmallError>
+        CORs policy won&apos;t allow fetching resources served over http from https, use a <code>https://</code> domain
```
Comment (pathmapper) [inline src/components/inputs/UrlInput.jsx]:
```suggestion
        CORS policy won&apos;t allow fetching resources served over http from https, use a <code>https://</code> domain
```

Or is the lower case `s` intentional?

Comment (orangemug) [reply src/components/inputs/UrlInput.jsx]: Nope you're completely right

**Comment by pathmapper** [comment]
Looks good! I like how the default protocol is set for the example URLs, very nice enhancement,

**Pull request merged**